### PR TITLE
Add note on special network values docs

### PR DESCRIPTION
**Pull request by DaveCTurner**
The special values `_global_`, `_site_`, `0.0.0.0` and so on may resolve
to multiple addresses, of which one is chosen to be the publish address.
This commit generalises the warning about reachability as applied to
DNS-resolved hostnames to also apply to these special values.

### CODE REVIEW

File: docs/reference/modules/network.asciidoc [hunk 1]
```diff
@@ -87,6 +87,11 @@ corresponding settings for the HTTP and transport interfaces.
 `0.0.0.0`::
   The addresses of all available network interfaces.
 
+NOTE: In some systems these special values resolve to multiple addresses. If
+so, {es} will select one of them as its publish address and may change its
+selection on each restart. You must make sure that your node is accessible at
+every possible address.
```
Comment (jrodewig) [inline docs/reference/modules/network.asciidoc:93]: I'd just make this last sentence imperative.
```suggestion
selection on each node restart. Ensure your node is accessible at every possible
address.
```

File: docs/reference/modules/network.asciidoc [hunk 2]
```diff
@@ -181,11 +186,12 @@ if binding to multiple addresses or using different addresses for publishing
 and binding.
 
 NOTE: You can specify a list of addresses for `network.host` and
-`network.publish_host`. You can also specify a single hostname which resolves
-to multiple addresses. If you do this then {es} chooses one of the addresses
-for its publish address. This choice uses heuristics based on IPv4/IPv6 stack
-preference and reachability and may change when the node restarts. You must
-make sure that each node is accessible at all possible publish addresses.
+`network.publish_host`. You can also specify one or more hostnames or
+<<network-interface-values,special values>> which resolve to multiple addresses
```
Comment (jrodewig) [inline docs/reference/modules/network.asciidoc:190]: Missing period. I'd also use `that` here.
```suggestion
<<network-interface-values,special values>> that resolve to multiple addresses.
```

File: docs/reference/modules/network.asciidoc [hunk 3]
```diff
@@ -181,11 +186,12 @@ if binding to multiple addresses or using different addresses for publishing
...
+If you do this then {es} chooses one of the addresses for its publish address.
+This choice uses heuristics based on IPv4/IPv6 stack preference and
+reachability and may change when the node restarts. You must make sure that
+each node is accessible at all possible publish addresses.
```
Comment (jrodewig) [inline docs/reference/modules/network.asciidoc:194]: Same as above. I'd change to imperative.
```suggestion
reachability and may change when the node restarts. Ensure
each node is accessible at all possible publish addresses.
```

**Review by jrodewig — APPROVED** [review]
LGTM aside from some non-blocking nits. Thanks @DaveCTurner.

**Comment by DaveCTurner** [comment]
Good suggestions as always, thanks James

**Comment by DaveCTurner** [comment]
@elasticmachine update branch

**Pull request merged**